src/metabase/lib/convert.cljc

```diff
@@ -352,17 +352,18 @@
   `:effective-type`, which is not used in options maps in legacy MBQL."
   [m]
   (not-empty
-   (into {}
-         (comp (disqualify)
-               (remove (fn [[k _v]]
-                         (= k :effective-type))))
-         ;; Following construct ensures that transformation mbql -> pmbql -> mbql, does not add base-type where those
-         ;; were not present originally. Base types are adeed in [[metabase.lib.query/add-types-to-fields]].
-         (if (contains? m :metabase.lib.query/transformation-added-base-type)
-           (dissoc m
-                   :metabase.lib.query/transformation-added-base-type
-                   :base-type)
-           m))))
+   (-> (into {}
+             (comp (disqualify)
+                   (remove (fn [[k _v]]
+                             (= k :effective-type))))
+             ;; Following construct ensures that transformation mbql -> pmbql -> mbql, does not add base-type where
+             ;; those were not present originally. Base types are adeed in [[metabase.lib.query/add-types-to-fields]].
+             (if (contains? m :metabase.lib.query/transformation-added-base-type)
+               (dissoc m
+                       :metabase.lib.query/transformation-added-base-type
+                       :base-type)
+               m))
+       (m/assoc-some :original-temporal-unit (:metabase.lib.field/original-temporal-unit m)))))
```

Comment:
Understood. Actually I had this exact question and forgot to ask it.

I guess it should it be relatively safe adding a new key (as opposed to removing or changing the shape of an existing key), but was wondering if it's something that has caused us problems in the past (updating legacy format), or just general uneasiness that it could cause problems.

Reply:
General uneasiness. When we do the conversion to legacy, we carefully remove some keys (all metabase lib fields), so it's clearly not totally open. Now, I think that's for the FE, and this property is not supposed to show up there, but still, I wanted to know how Cam sees these things.